### ISSUE TITLE
Metro OOMs

### ISSUE THREAD

**Issue by vemv** [status-im/status-react#3525]
Hi!

Newbie to Status development here, trying to build the project (iOS + iPhone). I can get past most steps, but I'm hitting the following OOM:

![image](https://user-images.githubusercontent.com/1162994/37043493-05630f8a-2161-11e8-87c9-e7d7a4dade78.png)

Should a `max-old-space-size` flag be placed somewhere? Not yet very familiar with the RN stack myself...

Btw I'm using a 8GB macbook, not sure if the project will run at all for such a machine. It's easy for cljs projects to take 4.5GB, so if this project (being more meaty than the average) took e.g. 6GB minimum, it wouldn't be surprising if the project OOMs given one always has other stuff running (Chrome, Slack etc).

Cheers - Victor

**Comment by mandrigin** [comment]
@vemv as a workaround you can use `node --max-old-space-size=4096 "./node_modules/react-native/local-cli/cli.js" start` to start the metro bundler instead of `make react-native`. That should suffice.

**Comment by mandrigin** [comment]
for the record, I have this issue as well, would be glad to have any guidance on how to avoid it 🤓

**Comment by vemv** [comment]
Thanks so far @mandrigin !

Now I'm seeing Metro getting stuck at 0%:

` BUNDLE  [ios, dev] ./index.ios.js ░░░░░░░░░░░░░░░░ 0.0% (0/1)`

Not sure if it's just a matter of waiting...

**Comment by mandrigin** [comment]
@vemv it takes quite a while on my MBP

**Comment by vemv** [comment]
Tried it again and for me, the alternative command definitely hangs forever

**Comment by mandrigin** [comment]
@vemv hmm, you can try giving it more RAM and see what happens...

**Comment by vemv** [comment]
Tried several times, didn't work

**Comment by dmitryn** [comment]
I've ran into this.
@vemv what's inside your `index.ios.js`? Should be just bunch of `require` statements with `start()` fn call on the end. More info on `index.*.js` files here: https://github.com/drapanjanas/re-natal/blob/7094b20a06fbbe23b91e92f8bf14762ac5b7cc3e/README.md#development-with-figwheel

<blockquote><img src="https://avatars3.githubusercontent.com/u/1648448?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/drapanjanas/re-natal">drapanjanas/re-natal</a></strong></div><div>re-natal - Bootstrap ClojureScript React Native apps</div></blockquote>

**Comment by vemv** [comment]
@dmitryn thanks for the support! I'll store the tip in my mind.

Now I git cloned status-react again and followed https://wiki.status.im/Building_Status from scratch (armed with a cleaner MacOS/brew/npm installation, more familiarity, etc).

One specific pitfall I had formerly fallen into was https://github.com/status-im/status-react/issues/3359 . As a workaround, in my original attempt I had changed the `yoga` requirement in the Podfile.

Probably that was a mistake, which snowballed into deeper issues. Now using ` gem install cocoapods -v 1.3.1`, the documented build works as advertised!

Closing the issue, since I cannot reproduce it anymore.
